src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -42,6 +45,18 @@ public class BatchHandler : DatabaseRequestHandler
                 else
                     ThrowNotSupportedType(contentType);
 
+                for (int i = 0; i < command.ParsedCommands.Count; i++)
+                {
+                    if (command.ParsedCommands.Array[i + command.ParsedCommands.Offset].Type != CommandType.PUT)
+                        continue;
+
+                    if (command.ParsedCommands.Array[i + command.ParsedCommands.Offset].Id?.EndsWith("/") != true)
+                        continue;
+
+                    command.ParsedCommands.Array[i + command.ParsedCommands.Offset].Id =
+                        await ServerStore.GenerateClusterIdentityAsync(command.ParsedCommands.Array[i + command.ParsedCommands.Offset].Id, Database.Name);
```

Comment:
That would require the allocation of an additional structure for those indexes/positions to be returned outside `ParseAsync`. Would it be fine?

`ParseAsync` is already contains async machinery due to `await RefillParserBuffer` calls there